### ISSUE TITLE
TERRAFORM CRASH: When deploying a service using fastly provider

### ISSUE THREAD

**Issue by omata** [fastly/terraform-provider-fastly#289]
### Terraform Version

Terraform v0.12.28

### Affected Resource(s)

Please list the resources as a list, for example:

- module.fastly.fastly_service_v1.myapp

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files

You can download my project from: https://d.pr/f/QQ0lmB

_Please keep it confidential._

### Debug Output

https://d.pr/fIxDjK

### Panic Output

https://d.pr/6timJ2

### Expected Behavior

Deploy the Fastly service with provided parameters. This configuration deploys a service, domains, backends, logging endpoints, conditions and dictionary items.

### Actual Behavior

Terraform memory crash

### Steps to Reproduce

Just issued the command:

```bash
cd env/all && terraform apply -var-file="fastly.tfvars" -var env_prefix=pre
```

### Important Factoids

A plain Fastly service using a global API Key.

### References

I seek for it, but I didn't found anything related with it.

**Comment by phamann** [comment]
Hi @omata thank you very much for submitting this crash report with so much detail - it was very useful for identifying the issue. I can confirm that this is a genuine runtime panic which thankfully has an easy fix (#290).

We have recently refactored the code for s3 logging endpoints and it seems an error check was missed which could result it a nil pointer dereference when we subsequently try to read a property if the error was non-nil. 

Details of the crash:
1) The missing error check: https://github.com/terraform-providers/terraform-provider-fastly/blob/v0.18.0/fastly/block_fastly_service_v1_s3logging.go#L51
2) Your terraform log reported the crash on this line when the program attempts to read a value on the potentially nil struct:  https://github.com/terraform-providers/terraform-provider-fastly/blob/v0.18.0/fastly/block_fastly_service_v1_s3logging.go#L63
3) The error that should be reported to the user but wasn't caught: https://github.com/terraform-providers/terraform-provider-fastly/blob/v0.18.0/fastly/block_fastly_service_v1_s3logging.go#L282-L288 

### Fix:
As per the suppressed error message, It's most likely that your s3 logging definitions had either of the required credential fields missing `s3_access_key` or `s3_secret_key`. While we release the fix for this crash, you will be able to fix the error locally by providing those fields in your `s3logging` blocks. Hope that helps.

I will update this issue once we have merged and released the real fix. Once agin thank you for reporting this issue.

**Comment by omata** [comment]
Thank you @phamann for a quick reply.

I will try your proposed fix and report if I have any aditional related comments.

Best regards,
